### PR TITLE
Tracing abstraction for Helidon

### PR DESCRIPTION
**Pull request by tomas-langer**
First step for #3717 

Abstraction to allow tracing usage backed by either Open Tracing, Open Telemetry (or any other tracing that supports the concept of spans).
This PR adds a Helidon specific abstraction, Open Tracing provider, implementation of opent tracing for Zipkin and Jaeger (as we had), and a wrapping API for OpenTelemetry, that allows manual setup of OpenTelemetry and Tracer and their registration with WebServer and other Helidon components.
In case of MP, the OpenTelemetry module will look for global `OpenTelemetry` instance and attempt to create a tracer from it.
Note for MP: MicroProfile Tracing specification WILL NOT WORK with Open Telemetry provider, as it is fully dependant on OpenTracing (we may improve this in the future)

### CODE REVIEW

File: microprofile/tracing/pom.xml [hunk 1]
```diff
@@ -38,7 +38,11 @@
         <dependency>
             <groupId>jakarta.enterprise</groupId>
             <artifactId>jakarta.enterprise.cdi-api</artifactId>
-            <scope>provided</scope>
+            <scope>prmnqovided</scope>
```
Comment (danielkec) [inline microprofile/tracing/pom.xml]: typo

File: tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java [hunk 2]
```diff
@@ -82,20 +84,17 @@ private static CompletionStage<WebServer> startIt(Config config, WebServer.Build
         // print a message at shutdown. If unsuccessful, print the exception.
         CompletionStage<WebServer> start = webServer.start();
 
-        start.thenAccept(ws -> {
+        return webServer.start()
```
Comment (danielkec) [inline tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java:87]: double start

File: tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java [hunk 3]
```diff
@@ -82,20 +84,17 @@ private static CompletionStage<WebServer> startIt(Config config, WebServer.Build
...
                     System.out.println(
                             "WEB server is up! http://localhost:" + ws.port() + "/greet");
                     ws.whenShutdown().thenRun(() -> System.out.println("WEB server is DOWN. Good bye!"));
                 })
-                .exceptionally(t -> {
+                .onError(t -> {
                     System.err.println("Startup failed: " + t.getMessage());
                     t.printStackTrace(System.err);
-                    return null;
                 });
```
Comment (danielkec) [inline tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java:97]: orError doesn't get executed if single is ignored

File: microprofile/tracing/pom.xml (hunk 1 above)
Comment (spericas) [inline microprofile/tracing/pom.xml]: What sort of scope this? :)

Comment (spericas) [reply microprofile/tracing/pom.xml]: Resolved now

Comment (tomas-langer) [reply microprofile/tracing/pom.xml]: That was a scope where you accidently write random text and then miss it ;)

File: tracing/tracing/src/main/java/io/helidon/tracing/MapHeaderConsumer.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public Optional<String> get(String key) {
+        return Optional.ofNullable(headers.get(key))
+                .flatMap(it -> {
+                    if (it.isEmpty()) {
+                        return Optional.empty();
+                    }
+                    return Optional.of(it.get(0));
+                });
```
Comment (danielkec) [inline tracing/tracing/src/main/java/io/helidon/tracing/MapHeaderConsumer.java:42]: Could be:
```suggestion
                .map(List::stream)
                .flatMap(Stream::findFirst);
```

**Review by spericas — APPROVED** [review]
Long PR, but most of the updates are rather mechanical. Nice work!

File: tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java (hunk 3 above)
Comment (tomas-langer) [reply tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/Main.java:93]: Ignoring elements from main, await in test.

File: tracing/tracing/src/main/java/io/helidon/tracing/MapHeaderConsumer.java (hunk 4 above)
Comment (tomas-langer) [reply tracing/tracing/src/main/java/io/helidon/tracing/MapHeaderConsumer.java]: Used your suggestion

**Review by danielkec — DISMISSED** [review]

**Review by danielkec — APPROVED** [review]

**Review by ljamen — COMMENTED** [review]
LGTM, I checked Java docs for copyright and spelling/typos, reviewed Tracing Readme,md. No adoc in this PR.

**Pull request merged**